composer.json

```diff
@@ -67,6 +67,12 @@
 		"check-cs": [
 			"@php ./vendor/squizlabs/php_codesniffer/scripts/phpcs"
 		],
+		"premium-check-cs": [
+			"composer require yoast/yoastcs:~1.2.1 --update-with-dependencies --no-suggest --no-interaction",
```

Comment:
Don't think this should be in the script command, prefer to have it in the `.travis.yml`.
It will not revert the version after running this and this can be run locally, which will result in unwanted changes on the branch you are on.

It will also be easier to clean up when this is just set in the `composer.json`.

Reply:
I'm not sure I understand you correctly.

The reason this has been set up like this, is to allow devs to easily test local branches for Premium against the Premium CS requirements.
If this was just done in the Travis script, dev would never be able to locally CS-check Premium branches as PHPCS would just not run as the versions used are too different.

The second `require` effectively reverts the first which means that after the CS run, there should be no real changes to the `composer.lock` file (other than the lock hash which a file reset should take care off).

Reply:
I think the main issue I have with this is that the `premium-check-cs` is not doing similar things to `check-cs`.

Having a `premium-check-cs-1.2` (temporary) script which does the `composer require` and calling `premium-check-cs` after which the `composer require` is reset, would reduce confusion about what it should do.

This also introduces two additional version numbers to keep up to date.
Would it be possible to do a `git checkout composer.*` instead of a `require` with version `~0.4.3`?

Reply:
> Having a premium-check-cs-1.2 (temporary) script

I see your point, but...
* Considering how much still needs to be done to get Free to be able to move over, I'm not sure how "temporary" that would be.
* Even after Free has moved over, we should probably keep the `premium-check-cs` script as the directories for which a different `testVersion` is needed are different between Premium vs Free.

> Would it be possible to do a git checkout composer.* instead of a require with version ~0.4.3?

Could you expand on that a bit more ? I'm not sure what you mean with `git checkout composer.*` and what that's supposed to do ?